---
- id: ossindex:pypi:pyftpdlib:CVE-2007-6736
  title: '[CVE-2007-6736]  Improper Limitation of a Pathname to a Restricted Directory
    ("Path Traversal")'
  description: Multiple directory traversal vulnerabilities in FTPServer.py in pyftpdlib
    before 0.2.0 allow remote authenticated users to access arbitrary files and directories
    via a .. (dot dot) in a (1) LIST, (2) STOR, or (3) RETR command.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  package_manager: pypi
  cve:
  - CVE-2007-6736
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=9
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=16
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn16&r=16&format=side&path=/trunk/pyftpdlib/FTPServer.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2007-6737
  title: "[CVE-2007-6737]  Improper Authentication"
  description: FTPServer.py in pyftpdlib before 0.2.0 does not increment the attempted_logins
    count for a USER command that specifies an invalid username, which makes it easier
    for remote attackers to obtain access via a brute-force attack.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  package_manager: pypi
  cve:
  - CVE-2007-6737
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=20
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=23
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn23&r=23&format=side&path=/trunk/pyftpdlib/FTPServer.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2007-6738
  title: "[CVE-2007-6738] pyftpdlib before 0.1.1 does not choose a random value for
    the port associated wi..."
  description: pyftpdlib before 0.1.1 does not choose a random value for the port
    associated with the PASV command, which makes it easier for remote attackers to
    obtain potentially sensitive information about the number of in-progress data
    connections by reading the response to this command.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  package_manager: pypi
  cve:
  - CVE-2007-6738
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2007-6739
  title: "[CVE-2007-6739]  Improper Input Validation"
  description: FTPServer.py in pyftpdlib before 0.2.0 allows remote attackers to cause
    a denial of service via a long command.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  package_manager: pypi
  cve:
  - CVE-2007-6739
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=3
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=20
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn20&r=20&format=side&path=/trunk/pyftpdlib/FTPServer.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2007-6740
  title: "[CVE-2007-6740]  Permissions, Privileges, and Access Controls"
  description: The ftp_STOU function in FTPServer.py in pyftpdlib before 0.2.0 does
    not limit the number of attempts to discover a unique filename, which might allow
    remote authenticated users to cause a denial of service via a STOU command.
  affected_package: pyftpdlib
  vulnerable_versions:
  - 0.1.1
  - '0.1'
  package_manager: pypi
  cve:
  - CVE-2007-6740
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=25
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=37
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn37&r=37&format=side&path=/trunk/pyftpdlib/FTPServer.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2007-6741
  title: "[CVE-2007-6741]  Permissions, Privileges, and Access Controls"
  description: The ftp_PORT function in FTPServer.py in pyftpdlib before 0.2.0 does
    not prevent TCP connections to privileged ports if the destination IP address
    matches the source IP address of the connection from the FTP client, which might
    allow remote authenticated users to conduct FTP bounce attacks via crafted FTP
    data, as demonstrated by an FTP bounce attack against a NAT server, a related
    issue to CVE-1999-0017.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  package_manager: pypi
  cve:
  - CVE-2007-6741
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=11
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=32
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn32&r=32&format=side&path=/trunk/pyftpdlib/FTPServer.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2008-7262
  title: '[CVE-2008-7262]  Improper Limitation of a Pathname to a Restricted Directory
    ("Path Traversal")'
  description: Multiple directory traversal vulnerabilities in FTPServer.py in pyftpdlib
    before 0.3.0 allow remote authenticated users to access arbitrary files and directories
    via vectors involving a symlink in a pathname to a (1) CWD, (2) DELE, (3) STOR,
    or (4) RETR command.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  - 0.2.0
  package_manager: pypi
  cve:
  - CVE-2008-7262
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=55
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2008-7263
  title: "[CVE-2008-7263]  Improper Authentication"
  description: ftpserver.py in pyftpdlib before 0.5.0 does not delay its response
    after receiving an invalid login attempt, which makes it easier for remote attackers
    to obtain access via a brute-force attack.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  - 0.2.0
  - 0.3.0
  - 0.4.0
  package_manager: pypi
  cve:
  - CVE-2008-7263
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=73
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=348
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn348&r=348&format=side&path=/trunk/pyftpdlib/ftpserver.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2008-7264
  title: "[CVE-2008-7264]  Improper Input Validation"
  description: The ftp_QUIT function in ftpserver.py in pyftpdlib before 0.5.0 allows
    remote authenticated users to cause a denial of service (file descriptor exhaustion
    and daemon outage) by sending a QUIT command during a disallowed data-transfer
    attempt.
  affected_package: pyftpdlib
  vulnerable_versions:
  - 0.3.0
  - '0.1'
  - 0.4.0
  - 0.1.1
  - 0.2.0
  package_manager: pypi
  cve:
  - CVE-2008-7264
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=71
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=344
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn344&r=344&format=side&path=/trunk/pyftpdlib/ftpserver.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2009-5010
  title: '[CVE-2009-5010]  Concurrent Execution using Shared Resource with Improper
    Synchronization ("Race Condition")'
  description: Race condition in the FTPHandler class in ftpserver.py in pyftpdlib
    before 0.5.1 allows remote attackers to cause a denial of service (daemon outage)
    by establishing and then immediately closing a TCP connection, leading to the
    accept function having an unexpected return value of None, a different vulnerability
    than CVE-2010-3494.
  affected_package: pyftpdlib
  vulnerable_versions:
  - 0.5.0
  - '0.1'
  - 0.1.1
  - 0.2.0
  - 0.3.0
  - 0.4.0
  package_manager: pypi
  cve:
  - CVE-2009-5010
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://bugs.python.org/issue6706
  - http://code.google.com/p/pyftpdlib/issues/detail?id=91
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=439
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn439&r=439&format=side&path=/trunk/pyftpdlib/ftpserver.py
  - http://www.openwall.com/lists/oss-security/2010/09/09/6
  - http://www.openwall.com/lists/oss-security/2010/09/11/2
  - http://www.openwall.com/lists/oss-security/2010/09/22/3
  - http://www.openwall.com/lists/oss-security/2010/09/24/3
  - https://bugs.launchpad.net/zodb/%20bug/135108
  - https://bugs.launchpad.net/zodb/+bug/135108
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2009-5011
  title: '[CVE-2009-5011]  Concurrent Execution using Shared Resource with Improper
    Synchronization ("Race Condition")'
  description: Race condition in the FTPHandler class in ftpserver.py in pyftpdlib
    before 0.5.2 allows remote attackers to cause a denial of service (daemon outage)
    by establishing and then immediately closing a TCP connection, leading to the
    getpeername function having an ENOTCONN error, a different vulnerability than
    CVE-2010-3494.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  - 0.2.0
  - 0.3.0
  - 0.4.0
  - 0.5.0
  - 0.5.1
  package_manager: pypi
  cve:
  - CVE-2009-5011
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=100
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=543
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn543&r=543&format=side&path=/trunk/pyftpdlib/ftpserver.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2009-5012
  title: "[CVE-2009-5012]  Permissions, Privileges, and Access Controls"
  description: ftpserver.py in pyftpdlib before 0.5.2 does not require the l permission
    for the MLST command, which allows remote authenticated users to bypass intended
    access restrictions and list the root directory via an FTP session.
  affected_package: pyftpdlib
  vulnerable_versions:
  - '0.1'
  - 0.1.1
  - 0.2.0
  - 0.3.0
  - 0.4.0
  - 0.5.0
  - 0.5.1
  package_manager: pypi
  cve:
  - CVE-2009-5012
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=114
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=596
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn596&r=596&format=side&path=/trunk/pyftpdlib/ftpserver.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2009-5013
  title: "[CVE-2009-5013]  Resource Management Errors"
  description: Memory leak in the on_dtp_close function in ftpserver.py in pyftpdlib
    before 0.5.2 allows remote authenticated users to cause a denial of service (memory
    consumption) by sending a QUIT command during a data transfer.
  affected_package: pyftpdlib
  vulnerable_versions:
  - 0.1.1
  - 0.2.0
  - 0.3.0
  - 0.4.0
  - 0.5.0
  - 0.5.1
  - '0.1'
  package_manager: pypi
  cve:
  - CVE-2009-5013
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://code.google.com/p/pyftpdlib/issues/detail?id=119
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=615
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn615&r=615&format=side&path=/trunk/pyftpdlib/ftpserver.py
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: ossindex:pypi:pyftpdlib:CVE-2010-3494
  title: '[CVE-2010-3494]  Concurrent Execution using Shared Resource with Improper
    Synchronization ("Race Condition")'
  description: Race condition in the FTPHandler class in ftpserver.py in pyftpdlib
    before 0.5.2 allows remote attackers to cause a denial of service (daemon outage)
    by establishing and then immediately closing a TCP connection, leading to the
    accept function having an unexpected value of None for the address, or an ECONNABORTED,
    EAGAIN, or EWOULDBLOCK error, a related issue to CVE-2010-3492.
  affected_package: pyftpdlib
  vulnerable_versions:
  - 0.1.1
  - 0.2.0
  - 0.3.0
  - 0.4.0
  - 0.5.0
  - 0.5.1
  - '0.1'
  package_manager: pypi
  cve:
  - CVE-2010-3494
  disclosed_date: 2010-10-19
  created_date: 2010-10-19
  last_modified_date: 2017-04-21
  credit:
  - OSSIndex
  references:
  - http://bugs.python.org/issue6706
  - http://code.google.com/p/pyftpdlib/issues/detail?id=104
  - http://code.google.com/p/pyftpdlib/issues/detail?id=105
  - http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY
  - http://code.google.com/p/pyftpdlib/source/detail?r=556
  - http://code.google.com/p/pyftpdlib/source/diff?spec=svn556&r=556&format=side&path=/trunk/pyftpdlib/ftpserver.py
  - http://www.openwall.com/lists/oss-security/2010/09/09/6
  - http://www.openwall.com/lists/oss-security/2010/09/11/2
  - http://www.openwall.com/lists/oss-security/2010/09/22/3
  - http://www.openwall.com/lists/oss-security/2010/09/24/3
  - https://bugs.launchpad.net/zodb/%20bug/135108
  - https://bugs.launchpad.net/zodb/+bug/135108
  source_url: https://ossindex.net/resource/package/8401202174/vulnerabilities
- id: snykio:pypi:pyftpdlib:CVE-2007-6736
  title: Directory Traversal
  description: |
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Directory Traversal
    attacks. A remote authenticated users to access arbitrary files and
    directories.



    [1]: https://pypi.org/project/pyftpdlib/
    \nA Directory Traversal attack (also known as path traversal) aims to
    access files and directories that are stored outside the intended
    folder. By manipulating files with \"dot-dot-slash (../)\" sequences and
    its variations, or by using absolute file paths, it may be possible to
    access arbitrary files and directories stored on file system, including
    application source code, configuration, and other critical system files.

    Directory Traversal vulnerabilities can be generally divided into two
    types:

    * **Information Disclosure**\: Allows the attacker to gain information
      about the folder structure or read the contents of sensitive files on
      the system.

    `st` is a module for serving static files on web pages, and contains a
    [vulnerability of this type][1]. In our example, we will serve files
    from the `public` route.

    If an attacker requests the following URL from our server, it will in
    turn leak the sensitive private key of the root user.

        curl http://localhost:8080/public/%2e%2e/%2e%2e/%2e%2e/%2e%2e/%2e%2e/root/.ssh/id_rsa

    **Note** `%2e` is the URL encoded version of `.` (dot).

    * **Writing arbitrary files**\: Allows the attacker to create or replace
      existing files. This type of vulnerability is also known as
      `Zip-Slip`.

    One way to achieve this is by using a malicious `zip` archive that holds
    path traversal filenames. When each filename in the zip archive gets
    concatenated to the target extraction folder, without validation, the
    final path ends up outside of the target folder. If an executable or a
    configuration file is overwritten with a file containing malicious code,
    the problem can turn into an arbitrary code execution issue quite
    easily.

    The following is an example of a `zip` archive with one benign file and
    one malicious file. Extracting the malicious file will result in
    traversing out of the target folder, ending up in `/root/.ssh/`
    overwriting the `authorized_keys` file:

        2018-04-15 22:04:29 .....           19           19  good.txt
        2018-04-15 22:04:42 .....           20           20  ../../../../../../root/.ssh/authorized_keys



    [1]: https://snyk.io/vuln/npm:st:20140206
  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.2.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2007-6736
  cwe:
  - CWE-22
  disclosed_date: 2007-06-15
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/9
  - https://nvd.nist.gov/vuln/detail/CVE-2007-6736
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42147
- id: snykio:pypi:pyftpdlib:CVE-2007-6737
  title: Access Restriction Bypass
  description: |
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Access Restriction
    Bypass. It does not increment the `attempted_logins` count for a USER
    command that specifies an invalid username, which makes it easier for
    remote attackers to obtain access via a brute-force attack.



    [1]: https://pypi.org/project/pyftpdlib/
  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.2.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2007-6737
  cwe:
  - CWE-287
  disclosed_date: 2007-07-13
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/20
  - https://nvd.nist.gov/vuln/detail/CVE-2007-6737
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42135
- id: snykio:pypi:pyftpdlib:CVE-2007-6739
  title: Denial of Service (DoS)
  description: |+
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Denial of Service
    (DoS) attacks via a long command.



    [1]: https://pypi.org/project/pyftpdlib/
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.2.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2007-6739
  cwe:
  - CWE-20
  disclosed_date: 2007-05-28
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/3
  - https://nvd.nist.gov/vuln/detail/CVE-2007-6739
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42137
- id: snykio:pypi:pyftpdlib:CVE-2007-6740
  title: Denial of Service (DoS)
  description: |+
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Denial of Service
    Attacks via the `ftp_STOU` function. It does not limit the number of
    attempts to discover a unique filename.



    [1]: https://pypi.org/project/pyftpdlib/
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.2.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2007-6740
  cwe:
  - CWE-264
  disclosed_date: 2007-07-18
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/25
  - https://nvd.nist.gov/vuln/detail/CVE-2007-6740
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42145
- id: snykio:pypi:pyftpdlib:CVE-2007-6741
  title: Man-in-the-Middle (MitM)
  description: |
    [pyftpdlib][1] is a Python FTP server library.

    Affected versions of this package are vulnerable to Man-in-the-Middle
    (MitM) attacks. It does not prevent `TCP` connections to privileged
    ports if the destination IP address matches the source IP address of the
    connection from the `FTP` client, which might allow remote authenticated
    users to conduct `FTP` bounce attacks via crafted `FTP` data.



    [1]: https://pypi.org/project/pyftpdlib/
  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.2.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2007-6741
  cwe:
  - CWE-264
  disclosed_date: 2007-06-18
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/11
  - https://nvd.nist.gov/vuln/detail/CVE-2007-6741
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42140
- id: snykio:pypi:pyftpdlib:CVE-2008-7262
  title: Directory Traversal
  description: |
    [pyftpdlib][1] is a Python FTP server library.

    Affected versions of this package are vulnerable to Directory Traversal
    attacks. A remote authenticated user could access arbitrary files and
    directories via vectors involving a symlink in a pathname.



    [1]: https://pypi.org/project/pyftpdlib/
    \nA Directory Traversal attack (also known as path traversal) aims to
    access files and directories that are stored outside the intended
    folder. By manipulating files with \"dot-dot-slash (../)\" sequences and
    its variations, or by using absolute file paths, it may be possible to
    access arbitrary files and directories stored on file system, including
    application source code, configuration, and other critical system files.

    Directory Traversal vulnerabilities can be generally divided into two
    types:

    * **Information Disclosure**\: Allows the attacker to gain information
      about the folder structure or read the contents of sensitive files on
      the system.

    `st` is a module for serving static files on web pages, and contains a
    [vulnerability of this type][1]. In our example, we will serve files
    from the `public` route.

    If an attacker requests the following URL from our server, it will in
    turn leak the sensitive private key of the root user.

        curl http://localhost:8080/public/%2e%2e/%2e%2e/%2e%2e/%2e%2e/%2e%2e/root/.ssh/id_rsa

    **Note** `%2e` is the URL encoded version of `.` (dot).

    * **Writing arbitrary files**\: Allows the attacker to create or replace
      existing files. This type of vulnerability is also known as
      `Zip-Slip`.

    One way to achieve this is by using a malicious `zip` archive that holds
    path traversal filenames. When each filename in the zip archive gets
    concatenated to the target extraction folder, without validation, the
    final path ends up outside of the target folder. If an executable or a
    configuration file is overwritten with a file containing malicious code,
    the problem can turn into an arbitrary code execution issue quite
    easily.

    The following is an example of a `zip` archive with one benign file and
    one malicious file. Extracting the malicious file will result in
    traversing out of the target folder, ending up in `/root/.ssh/`
    overwriting the `authorized_keys` file:

        2018-04-15 22:04:29 .....           19           19  good.txt
        2018-04-15 22:04:42 .....           20           20  ../../../../../../root/.ssh/authorized_keys



    [1]: https://snyk.io/vuln/npm:st:20140206
  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.3.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2008-7262
  cwe:
  - CWE-22
  disclosed_date: 2008-11-26
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/55
  - https://nvd.nist.gov/vuln/detail/CVE-2008-7262
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42136
- id: snykio:pypi:pyftpdlib:CVE-2008-7263
  title: Access Restriction Bypass
  description: |
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Access Restriction
    Bypass. It does not delay its response after receiving an invalid login
    attempt, which makes it easier for remote attackers to obtain access via
    a brute-force attack.



    [1]: https://pypi.org/project/pyftpdlib/
  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.0"
  severity: high
  package_manager: pypi
  cve:
  - CVE-2008-7263
  cwe:
  - CWE-287
  disclosed_date: 2008-07-14
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/73
  - https://nvd.nist.gov/vuln/detail/CVE-2008-7263
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42138
- id: snykio:pypi:pyftpdlib:CVE-2008-7264
  title: Denial of Service (DoS)
  description: |+
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Denial of Service
    (DoS) attacks via the `ftp_QUIT` function, by sending a `QUIT` command
    during a disallowed data-transfer attempt.



    [1]: https://pypi.org/project/pyftpdlib/
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2008-7264
  cwe:
  - CWE-20
  disclosed_date: 2008-06-26
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/71
  - https://nvd.nist.gov/vuln/detail/CVE-2008-7264
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42143
- id: snykio:pypi:pyftpdlib:CVE-2009-5010
  title: Denial of Service (DoS)
  description: |+
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Denial of Service
    (DoS) attacks via the `FTPHandler` class.



    [1]: https://pypi.org/project/pyftpdlib/
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.1"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2009-5010
  cwe:
  - CWE-362
  disclosed_date: 2008-11-17
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/91
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42141
- id: snykio:pypi:pyftpdlib:CVE-2009-5011
  title: Denial of Service (DoS)
  description: |+
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Denial of Service
    Attacks. A Malicious user could cause a denial of service by
    establishing and then immediately closing a TCP connection, leading to
    the `getpeername` function having an `ENOTCONN` error.



    [1]: https://pypi.org/project/pyftpdlib/
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.2"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2009-5011
  cwe:
  - CWE-362
  disclosed_date: 2009-02-27
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://nvd.nist.gov/vuln/detail/CVE-2009-5011
  - https://github.com/giampaolo/pyftpdlib/issues/100
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42139
- id: snykio:pypi:pyftpdlib:CVE-2009-5012
  title: Access Restriction Bypass
  description: |
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to Access Restriction
    Bypass. It does not require the `l` permission for the `MLST` command,
    which allows remote authenticated users to bypass intended access
    restrictions and list the root directory via an`FTP` session.



    [1]: https://pypi.org/project/pyftpdlib/
  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.2"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2009-5012
  cwe:
  - CWE-264
  disclosed_date: 2009-04-21
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/114
  - https://nvd.nist.gov/vuln/detail/CVE-2009-5012
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42144
- id: snykio:pypi:pyftpdlib:CVE-2009-5013
  title: Denial of Service (DoS)
  description: |+
    [pyftpdlib][1] is a FTP server library.

    Affected versions of this package are vulnerable to high memory
    consumption Denial of Service (DoS) by sending a `QUIT` command during a
    data transfer.



    [1]: https://pypi.org/project/pyftpdlib/
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.2"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2009-5013
  cwe:
  - CWE-399
  disclosed_date: 2009-07-29
  created_date: 2018-06-04
  last_modified_date: 2018-06-04
  credit:
  - Unknown
  references:
  - https://github.com/giampaolo/pyftpdlib/issues/119
  - https://nvd.nist.gov/vuln/detail/CVE-2009-5013
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-42146
- id: snykio:pypi:pyftpdlib:CVE-2010-3494
  title: Denial of Service (DoS)
  description: |+
    [`pyftpdlib`][1] is a Very fast asynchronous FTP server library.

    Race condition in the FTPHandler class in ftpserver.py in pyftpdlib
    before 0.5.2 allows remote attackers to cause a denial of service
    (daemon outage) by establishing and then immediately closing a TCP
    connection, leading to the accept function having an unexpected value of
    None for the address, or an ECONNABORTED, EAGAIN, or EWOULDBLOCK error,
    a related issue to CVE-2010-3492.



    [1]: https://pypi.python.org/pypi/pyftpdlib
    \nDenial of Service (DoS) describes a family of attacks, all aimed at
    making a system inaccessible to its intended and legitimate users.

    Unlike other vulnerabilities, DoS attacks usually do not aim at
    breaching security. Rather, they are focused on making websites and
    services unavailable to genuine users resulting in downtime.

    One popular Denial of Service vulnerability is DDoS (a Distributed
    Denial of Service), an attack that attempts to clog network pipes to the
    system by generating a large volume of traffic from many machines.

    When it comes to open source libraries, DoS vulnerabilities allow
    attackers to trigger such a crash or crippling of the service by using a
    flaw either in the application code or from the use of open source
    libraries.

    Two common types of DoS vulnerabilities:

    * High CPU/Memory Consumption- An attacker sending crafted requests that
      could cause the system to take a disproportionate amount of time to
      process. For example,
      [commons-fileupload:commons-fileupload](SNYK-JAVA-COMMONSFILEUPLOAD-30082).

    * Crash - An attacker sending crafted requests that could cause the
      system to crash. For Example, [npm `ws` package](npm:ws:20171108)

  affected_package: pyftpdlib
  vulnerable_versions:
  - "< 0.5.2"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2010-3494
  cwe:
  - CWE-362
  disclosed_date: 2009-04-02
  created_date: 2009-04-02
  last_modified_date: 2009-04-02
  credit:
  - Giampaolo Rodola
  references:
  - https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2010-3494
  - http://bugs.python.org/issue6706
  - https://github.com/giampaolo/pyftpdlib/issues/105
  - https://github.com/giampaolo/pyftpdlib/commit/0f822329e3431ec1bc6250c03e938c65a61b2eb4
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYFTPDLIB-40054
